### ISSUE TITLE
Local terminal left in broken state

### ISSUE THREAD

**Issue by hyperknot** [fabric/fabric#1907]
Many times, when I use Fabric 2, after the script terminates in an exception my local terminal breaks. It doesn't show any input text what I type. "Invisibly" the terminal still works, I can type blindly and it shows output.

macOS 10.13.6, iTerm 3.2.5, Python 3.7

**Comment by ploxiln** [comment]
not an excuse, but if you're interested in a way to fix the terminal when in such a state, try `<enter>reset<enter>` (just the `reset` command)

**Comment by hyperknot** [comment]
Yes, Ctrl + C to clear whatever might have been entered + `reset` works.

**Comment by dcbb** [comment]
Any fix on this? Any way to run reset in a `finally` block?

**Comment by wshayes** [comment]
Looks similar to https://github.com/pyinvoke/invoke/issues/303 which was resolved .
